gateleen-kafka/src/main/java/org/swisspush/gateleen/kafka/KafkaHandler.java

```diff
@@ -92,14 +93,25 @@ public boolean isInitialized() {
     private Future<Void> initializeKafkaConfiguration(Buffer configuration) {
         Future<Void> future = Future.future();
         String replacedConfig;
-        replacedConfig = StringUtils.replaceWildcardConfigs(configuration.toString(UTF_8), properties);
-        final List<KafkaConfiguration> kafkaConfigurations = KafkaConfigurationParser.parse(
-                Buffer.buffer(replacedConfig));
+        final List<KafkaConfiguration> kafkaConfigurations = new ArrayList<>();
+
+        try {
+            replacedConfig = StringUtils.replaceWildcardConfigs(configuration.toString(UTF_8), properties);
+            kafkaConfigurations.addAll(KafkaConfigurationParser.parse(Buffer.buffer(replacedConfig)));
+        } catch (Exception e) {
+            log.warn("Could not replace wildcards with environment properties for kafka configurations due to following reason: {}",
+                    e.getMessage());
+        }
+
         repository.closeAll().setHandler(event -> {
-            for (KafkaConfiguration kafkaConfiguration : kafkaConfigurations) {
-                repository.addKafkaProducer(kafkaConfiguration);
+            if (kafkaConfigurations.isEmpty()) {
+                initialized = false;
```

Comment:
empty configurations are not leading to an un-inititialized handler. I just noticed, that the wildcard replacing should be made in the `KafkaConfigurationParser` and not in the `KafkaHandler` -> separation of concerns (SoC)

Reply:
implementation moved to `KafkaConfigurationParser`